src/cir.rs

```diff
@@ -6,22 +6,9 @@ use crate::semantics::FieldCounter;
 use std::collections::HashMap;
 
 #[derive(Debug)]
-pub struct CirProgram {
-    pub(crate) instructions: Vec<CirInstruction>,
-}
-
-impl CirProgram {
-    pub fn new() -> Self {
-        Self {
-            instructions: Vec::new(),
-        }
-    }
-}
-
-impl Default for CirProgram {
-    fn default() -> Self {
-        Self::new()
-    }
+pub enum CirProgram {
+    Instructions(Vec<CirInstruction>),
```

Comment:
Should we use boxed slice instead `Vec` to emphasize that instructions won't grow over time?